[X Window System] add libraries for the x window systems

Neumann-A opened the pull request:
- main goal: compile xserver everywhere. (extracted from #9202)

will close the following issues
closes #6705

- [x] depends #15200
- [x] depends #15622
- [x] depends #15578

JackBoosY commented:
/azp run

Voskrese commented:
> -- Installing x64-windows-rel
> CMake Error at ports/xorg-macros/portfile.cmake:38 (file):
>   file RENAME failed to rename
> 
>     E:/tools/vcpkg/packages/xorg-macros_x64-windows/share/aclocal/
> 
>   to
> 
>     E:/tools/vcpkg/packages/xorg-macros_x64-windows/share/xorg/aclocal
> 
>   because: File exists
> 
> Call Stack (most recent call first):
>   scripts/ports.cmake:90 (include)

JackBoosY commented:
/azp run

JackBoosY commented:
/azp run

JackBoosY commented:
Oh, sorry! I click the ready for review buttom...
If this PR is ready for review, please ping me.

JackBoosY commented:
Pinging @Neumann-A  for response. Is work still being done for this PR?

Neumann-A commented:
@JackBoosY: This depends on #10402

Neumann-A committed:
[libpng] add pc file installation

Neumann-A committed:
[bzip2] install pc file and leave a TODO for somebody else ;)

Neumann-A committed:
[bzip2] bump control

Neumann-A committed:
WIP Commit 5/N

Neumann-A committed:
WIP Commit

Neumann-A committed:
wip commit

Neumann-A committed:
wip commit

Neumann-A committed:
xcursor fixes

Neumann-A committed:
more wip work

Neumann-A committed:
add mtdev

Neumann-A committed:
wip commit

Neumann-A committed:
try to install setuptools

Neumann-A committed:
add autoconfig to xorg-macros

Neumann-A committed:
wip commit

Neumann-A committed:
add missing xorg dep

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

Neumann-A committed:
WIP commit

Neumann-A committed:
try dbus again.

Neumann-A committed:
[vcpkg/scripts] allow to find internal meson.py

Neumann-A committed:
fix some issues

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

# Conflicts:
#	scripts/azure-pipelines/linux/provision-image.sh

Neumann-A committed:
[vcpkg/scripts/pkgconfig] skip checks if pkgconfig cannot be found.
- small bugfix in ignore flags regex

Neumann-A committed:
retry zstd and zlib

Neumann-A committed:
[libpng] retry by adding dl and m to system libraries for osx

Neumann-A committed:
[libpng] add missing vcpkg_fixup_pkgconfig()

Neumann-A committed:
[vcpkg/scripts/pkgconfig] unset var if found. 

fixes checks for other configurations which will be always succesful otherwise

Neumann-A committed:
[libpng] bump control

Neumann-A committed:
[libpng] enable port debug to debug osx regression

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into use_pkg_config_to_fixup_pkgconfig

# Conflicts:
#	scripts/azure-pipelines/osx/azure-pipelines.yml

Neumann-A committed:
ws change to retrigger osx ci and hopeing that it works

Neumann-A committed:
fix typo

Neumann-A committed:
Merge remote-tracking branch 'origin/use_pkg_config_to_fixup_pkgconfig' into add_x_window_system

# Conflicts:
#	ports/bzip2/bzip2.pc.in
#	ports/zlib/portfile.cmake
#	scripts/azure-pipelines/osx/azure-pipelines.yml
#	scripts/cmake/vcpkg_fixup_pkgconfig.cmake

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

Neumann-A committed:
[vcpkg_find_acquire_program] add version check for ninja

Neumann-A committed:
[fontconfig] update fontconfig

Neumann-A committed:
Change VERSION to NINJA_VERSION

Neumann-A committed:
Missed VERSION change

Neumann-A committed:
[libiconv] update to make

Neumann-A committed:
[gettext] update to make

Neumann-A committed:
[vcpkg_configure_make] Changes considering pure configure ports without the requirement to run autoconfig

Neumann-A committed:
[fontconfig] make it work on windows

Neumann-A committed:
some more corrections

Neumann-A committed:
add missing PACKAGES parameter

Neumann-A committed:
add osx hash

Co-authored-by: Wolfgang Stöggl <c72578@yahoo.de>

Neumann-A committed:
[meson] update to 0.55.1

Neumann-A committed:
fix command

Neumann-A committed:
Remove apt/brew package names

Co-authored-by: ras0219 <533828+ras0219@users.noreply.github.com>

Neumann-A committed:
move supported around and disable it for freebsd

Neumann-A committed:
fix small command hickup which does not matter for ninja

Neumann-A committed:
Merge branch 'ninja-version-check' into update_meson_0.55.1

# Conflicts:
#	scripts/cmake/vcpkg_find_acquire_program.cmake

Neumann-A committed:
Remove install of unofficial cmake config

Neumann-A committed:
add function get_cmake_vars

Neumann-A committed:
fine tuning.

Neumann-A committed:
apply to make based ports.

Neumann-A committed:
add log suffix on not windows platforms

Neumann-A committed:
fix c&p error

Neumann-A committed:
add previous LINK env

Neumann-A committed:
setup env on windows and extract cpp flags correctly.

Neumann-A committed:
update glib and libxml2

strega-nil committed:
Merge remote-tracking branch 'upstream/master' into update_meson_0.55.1

Neumann-A committed:
Merge remote-tracking branch 'origin/fix_script_details' into make_meson_sane

# Conflicts:
#	scripts/cmake/vcpkg_find_acquire_program.cmake

Neumann-A committed:
revert changes to acquire program

Neumann-A committed:
Merge remote-tracking branch 'origin/update_meson_0.55.1' into make_meson_sane

# Conflicts:
#	scripts/cmake/vcpkg_find_acquire_program.cmake

Neumann-A committed:
Merge remote-tracking branch 'origin/add_get_cmake_vars' into make_meson_sane

Neumann-A committed:
refactor configure meson

Neumann-A committed:
remove crossfile from triplet

Neumann-A committed:
fix c&p error

Neumann-A committed:
add compilers in cross file

Neumann-A committed:
fix windows regressions

Neumann-A committed:
Apply suggestions from code review

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into make_meson_sane

Neumann-A committed:
fix missing system name

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_fontconfig

# Conflicts:
#	ports/fontconfig/CMakeLists.txt
#	ports/fontconfig/CONTROL
#	ports/fontconfig/portfile.cmake

Neumann-A committed:
Merge remote-tracking branch 'origin/add_get_cmake_vars' into update_fontconfig

# Conflicts:
#	scripts/cmake/vcpkg_configure_make.cmake

Neumann-A committed:
add windres wrapper to invoke RC.

Neumann-A committed:
remove wrong fi

Neumann-A committed:
fix libintl.

Neumann-A committed:
try fixing uwp

Neumann-A committed:
other things to update

Neumann-A committed:
exiv2 fix

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_fontconfig

# Conflicts:
#	scripts/cmake/vcpkg_configure_make.cmake

Neumann-A committed:
libtool does not like -RTC1

Neumann-A committed:
Merge branch 'update_fontconfig' of https://github.com/Neumann-A/vcpkg into update_fontconfig

Neumann-A committed:
pass the /RTC<x> flag directly to the compiler

Neumann-A committed:
trust cmake instead of adding extra flags

Neumann-A committed:
fix expat

Neumann-A committed:
fix iconv wrapper install

Neumann-A committed:
change fontconfig intl linkage.

Neumann-A committed:
add appcontainer back in which is required for UWP. Why is this not in the cmake flags?

Neumann-A committed:
fix pkg-config in vcpkg_configure_make

Neumann-A committed:
fix json-c pkg-config installation

Neumann-A committed:
remove remnoved config

Neumann-A committed:
comment out debug messages + bit of tuning

Neumann-A committed:
finally fontconfig

Neumann-A committed:
comment debug messages

Neumann-A committed:
expat apply extra patches.

Neumann-A committed:
Switch back to -E instead of -EP in cpp

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_fontconfig

# Conflicts:
#	scripts/cmake/vcpkg_build_make.cmake

Neumann-A committed:
commit changes from fontconfig PR

Neumann-A committed:
Merge branch 'add_get_cmake_vars' of https://github.com/Neumann-A/vcpkg into add_get_cmake_vars

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_get_cmake_vars

# Conflicts:
#	scripts/cmake/vcpkg_build_make.cmake
#	scripts/cmake/vcpkg_configure_make.cmake

Neumann-A committed:
[expat] fix expat details

Neumann-A committed:
Apply suggestions from code review

Co-authored-by: nicole mazzuca <mazzucan@outlook.com>

Neumann-A committed:
cleanup docs

Neumann-A committed:
Merge remote-tracking branch 'origin/add_get_cmake_vars' into make_meson_sane

# Conflicts:
#	scripts/cmake/vcpkg_configure_meson.cmake

Neumann-A committed:
update osx pipeline so that fontconfig actually builds

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

# Conflicts:
#	ports/bzip2/CONTROL
#	ports/bzip2/portfile.cmake
#	ports/expat/CONTROL
#	ports/expat/portfile.cmake
#	ports/fontconfig/CONTROL
#	ports/freetype/portfile.cmake
#	ports/gmp/CONTROL
#	ports/libepoxy/CONTROL
#	ports/libepoxy/portfile.cmake
#	ports/libffi/portfile.cmake
#	ports/liblzma/CMakeLists.txt
#	ports/liblzma/CONTROL
#	ports/liblzma/portfile.cmake
#	ports/libpng/CONTROL
#	ports/libpng/pkgconfig.2.patch
#	ports/libpng/portfile.cmake
#	ports/opengl/CONTROL
#	ports/opengl/portfile.cmake
#	ports/zlib/CONTROL
#	ports/zlib/portfile.cmake
#	ports/zstd/CONTROL
#	ports/zstd/portfile.cmake
#	scripts/azure-pipelines/linux/provision-image.sh
#	scripts/azure-pipelines/osx/azure-pipelines.yml
#	scripts/cmake/vcpkg_build_make.cmake
#	scripts/cmake/vcpkg_configure_make.cmake
#	scripts/cmake/vcpkg_configure_meson.cmake
#	scripts/cmake/vcpkg_find_acquire_program.cmake
#	scripts/cmake/vcpkg_fixup_pkgconfig.cmake

Neumann-A committed:
Merge remote-tracking branch 'origin/make_meson_sane' into add_x_window_system

# Conflicts:
#	scripts/cmake/vcpkg_configure_meson.cmake

Neumann-A committed:
fix libxslt lzma linkage

Neumann-A committed:
fix build to xlib

Neumann-A committed:
fix expat the lazy way by using a higher commit than release

Neumann-A committed:
fix a barage of regressions due to the use of unofficial targets

Neumann-A committed:
[expat] use a higher commit hash with a better stabilized cmake build

Neumann-A committed:
forgot freexl regression

Neumann-A committed:
more unofficial fixes

Neumann-A committed:
fix fribidi regressions

Neumann-A committed:
fix downstream expat usage

Neumann-A committed:
fix downstream expat usage

Neumann-A committed:
- automatic rename of static libraries on windows
- move c_args and others into properties since builtin options wasn'T accepted
- fribidi add fixup_pkgconfig

Neumann-A committed:
remove fatal error

Neumann-A committed:
fix wxwidgets

Neumann-A committed:
fix gcdm regression

Neumann-A committed:
[vtk] fix the regression due to expat

Neumann-A committed:
Merge remote-tracking branch 'origin/fix_expat_details' into update_fontconfig

# Conflicts:
#	ports/expat/vcpkg-cmake-wrapper.cmake
#	ports/fontconfig/CMakeLists.txt
#	ports/fontconfig/CONTROL

Neumann-A committed:
add uwp cl flags back in. Somehow those are not set by cmake

Neumann-A committed:
add a few other configure options for full gettext build

Neumann-A committed:
fix cmake regression

Neumann-A committed:
fix a few regressions

Neumann-A committed:
fix static gettext build

Neumann-A committed:
fix gettext static

Neumann-A committed:
fix libxml2 wrapper for cmake

Neumann-A committed:
some more regressions fixes

Neumann-A committed:
add conversion from somelib.lib to -lsomelib

Neumann-A committed:
Merge remote-tracking branch 'origin/add_get_cmake_vars' into add_x_window_system

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

Neumann-A committed:
Merge remote-tracking branch 'origin/add_get_cmake_vars' into make_meson_sane

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into make_meson_sane

Neumann-A committed:
reset expat changes

Neumann-A committed:
remove cross toolchain

Neumann-A committed:
fix windows system libs

Neumann-A committed:
fix libxslt linux build

Neumann-A committed:
skip pc check in xproto

Neumann-A committed:
Merge remote-tracking branch 'origin/make_meson_sane' into add_x_window_system

Neumann-A committed:
add a few option to libiconv.

Neumann-A committed:
get logs from CI

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into update_fontconfig

# Conflicts:
#	ports/libtorrent/CONTROL

Neumann-A committed:
add missing ar-lib wrapper

Neumann-A committed:
add missing ar-lib wrapper

Neumann-A committed:
add fatal error to gettext

Neumann-A committed:
remove uuid from the list of LIBS since it seems to not exist in CI?

Neumann-A committed:
small but important regex correction

Neumann-A committed:
fix regex and add debug message for libs

Neumann-A committed:
remove error and only build libintl.

Neumann-A committed:
add uuid dependency to fontconfig in qt5-base

Neumann-A committed:
osx install gettext for autopoint

Neumann-A committed:
fix io2d regression by saying the port is broken .....
(which it is; upstream needs to learn proper cmake)

Neumann-A committed:
restore the old libxml2 wrapper with minor modifications

Neumann-A committed:
Merge branch 'update_fontconfig' of https://github.com/Neumann-A/vcpkg into update_fontconfig

Neumann-A committed:
fix xmlsec regression

Neumann-A committed:
install wrapper correctly

Neumann-A committed:
try actual fixing io2d

Neumann-A committed:
improve iconv wrapper

Neumann-A committed:
add latest changes from update_fontconfig PR

ras0219 committed:
[vcpkg_fixup_pkgconfig] Large simplification, handle spaces in path.

Add test for zlib with spaces in path.

ras0219 committed:
[vcpkg] Update spaces end-to-end test to also use capitals

ras0219 committed:
[vcpkg_fixup_pkgconfig] Improve error messages and add filesystem sleep shim

Neumann-A committed:
Merge remote-tracking branch 'origin/add_get_cmake_vars' into add_x_window_system

Neumann-A committed:
Merge remote-tracking branch 'origin/update_fontconfig' into add_x_window_system

# Conflicts:
#	ports/fontconfig/CMakeLists.txt
#	ports/fontconfig/portfile.cmake

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

Neumann-A committed:
Merge remote-tracking branch 'ras/dev/roschuma/pkgconfig-spaces' into add_x_window_system

ras0219-msft committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/roschuma/pkgconfig-spaces

ras0219-msft committed:
Merge remote-tracking branch 'origin/master' into dev/roschuma/pkgconfig-spaces

ras0219-msft committed:
[nettle] Fix pkgconfig generation on Windows

ras0219-msft committed:
[gmp] Handle change in install_msbuild

ras0219-msft committed:
[vcpkg_fixup_pkgconfig] Handle spaces on non-Windows platforms

ras0219 committed:
[vcpkg_fixup_pkgconfig] Handle issue with win32 pkg-config double escaping

BillyONeal committed:
Merge remote-tracking branch 'origin/master' into dev/roschuma/pkgconfig-spaces

ras0219 committed:
[vcpkg_fixup_pkgconfig] Ignore implicit libraries in .pc file (stdc++, gcc_s, etc)

ras0219-msft committed:
Merge branch 'master' of https://github.com/microsoft/vcpkg into dev/roschuma/pkgconfig-spaces

Neumann-A committed:
Merge remote-tracking branch 'ras/dev/roschuma/pkgconfig-spaces' into add_x_window_system

# Conflicts:
#	ports/expat/CONTROL
#	ports/expat/fix-find-package-by-cmake.patch
#	ports/expat/portfile.cmake
#	ports/expat/vcpkg-cmake-wrapper.cmake
#	ports/fontconfig/CMakeLists.txt
#	ports/fontconfig/CONTROL
#	ports/itk/expat.patch
#	ports/poco/use-vcpkg-expat.patch
#	ports/wxwidgets/fix-expat.patch

Neumann-A committed:
Update meson to a recent commit to fix a bug with 0.56

Neumann-A committed:
rename static libraries built by meson by default on windows

Neumann-A committed:
apply cmake variables detection to meson.
use native/cross files
make arm builds work.

Neumann-A committed:
update munit

Neumann-A committed:
fix fribidi

Neumann-A committed:
fix _FLAGS variables

Neumann-A committed:
set env SDKROOT and MACOSX_DEPLOYMENT_TARGET in configure_meson

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

# Conflicts:
#	ports/ffmpeg/FindFFMPEG.cmake.in
#	ports/fontconfig/portfile.cmake
#	ports/gdal/CONTROL
#	ports/gettext/portfile.cmake
#	ports/gmp/vcpkg.json
#	ports/harfbuzz/CONTROL
#	ports/io2d/portfile.cmake
#	ports/libepoxy/portfile.cmake
#	ports/libffi/portfile.cmake
#	ports/libiconv/portfile.cmake
#	ports/libspatialite/CONTROL
#	ports/libtorrent/portfile.cmake
#	ports/libxml2/CMakeLists.txt
#	ports/libxml2/CONTROL
#	ports/libxslt/portfile.cmake
#	ports/nettle/CONTROL
#	ports/nettle/portfile.cmake
#	ports/osg/CONTROL
#	ports/pixman/CONTROL
#	ports/pixman/portfile.cmake
#	ports/qt5-base/CONTROL
#	ports/spatialite-tools/CONTROL
#	scripts/azure-pipelines/end-to-end-tests.ps1
#	scripts/azure-pipelines/osx/azure-pipelines.yml
#	scripts/cmake/vcpkg_apply_patches.cmake
#	scripts/cmake/vcpkg_build_make.cmake
#	scripts/cmake/vcpkg_common_functions.cmake
#	scripts/cmake/vcpkg_configure_make.cmake
#	scripts/cmake/vcpkg_configure_meson.cmake
#	scripts/cmake/vcpkg_fixup_pkgconfig.cmake
#	scripts/get_cmake_vars/CMakeLists.txt

Neumann-A committed:
Merge remote-tracking branch 'origin/make_meson_sane' into add_x_window_system

# Conflicts:
#	scripts/cmake/vcpkg_install_meson.cmake

Neumann-A committed:
reset gmp portfile

Neumann-A committed:
reset scripts/azure-pipelines/linux/provision-image.sh

Neumann-A committed:
remove moved files

Neumann-A committed:
cleanup pr

Neumann-A committed:
more cleanup

Neumann-A committed:
need osx env while building and configuring

Neumann-A committed:
- remove unused env backup functions
- rename internal functions to be prefixed with _vcpkg_
- fix VCPKG_CHAINLOAD_TOOLCHAIN_FILE for ios/freebsd

Neumann-A committed:
replace _vcpkg with vcpkg since _ has special meaning in functions

Neumann-A committed:
rename functions to vcpkg_internal_meson_generate

Neumann-A committed:
update control tool-meson

Neumann-A committed:
Merge remote-tracking branch 'origin/make_meson_sane' into add_x_window_system

Neumann-A committed:
[mesa] new port

Neumann-A committed:
make mesa build more.

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into mesa

Neumann-A committed:
more fine tuning

Neumann-A committed:
fix x86

Neumann-A committed:
add supports.

Neumann-A committed:
comment about EGL feature.

PhoebeHui committed:
Resolve conficts

PhoebeHui committed:
Resolve conflicts

PhoebeHui committed:
Merge branch 'mesa' of https://github.com/Neumann-A/vcpkg into mesa

RichLogan committed:
[tool-meson] Rev vcpkg's meson build

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into mesa

Neumann-A committed:
apply suggestion from CR.
added opengl feature.
error on disabled shared glapi if multiple GL APIs are selected.

Neumann-A committed:
fix boolean option

Neumann-A committed:
make sharedgl-api=true the hardcoded default

Neumann-A committed:
add suggestion from CR

Neumann-A committed:
add mesa:x64-windows-static=fail to ci.baseline.txt due to static crt and parts being a dynamic library.

Neumann-A committed:
apply CR sugesstion

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

# Conflicts:
#	ports/elfutils/CONTROL
#	ports/elfutils/configure.ac.patch
#	ports/elfutils/portfile.cmake
#	ports/opengl/CONTROL
#	ports/x265/CONTROL
#	ports/x265/portfile.cmake
#	scripts/cmake/vcpkg_apply_patches.cmake
#	scripts/cmake/vcpkg_common_definitions.cmake
#	scripts/cmake/vcpkg_configure_meson.cmake

Neumann-A committed:
remove old hopefully unused stuff

Neumann-A committed:
Merge remote-tracking branch 'origin/mesa' into add_x_window_system

Neumann-A commented on ports/applewm/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+    REF 78e9081bd1dc88e4a86fc6e401647bdb5265fef7 # 1.4.1
+    SHA512  a4afb77458e980f61b0d51d8d89f89c2b47eaca6674b7ff2473745d04b9dd59921401db74f53fe24e5b5e4221f311f69fcbfcaa0d5e0b37c53e68a6844e655bc
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/applewm/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/dbus/portfile.cmake:
```diff
@@ -0,0 +1,54 @@
+        -DDBUS_ENABLE_XML_DOCS=OFF
+        -DDBUS_INSTALL_SYSTEM_LIBS=ON
+        -DXSLTPROC_EXECUTABLE=FALSE
+    #OPTIONS ${OPTIONS} --enable-tests=no
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/dbus/portfile.cmake:
```diff
@@ -0,0 +1,54 @@
+)
+vcpkg_install_cmake()
+vcpkg_fixup_cmake_targets(CONFIG_PATH "lib/cmake/DBus1")
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES gcc gcc_s c pthread /usr/lib/x86_64-linux-gnu/librt.so SKIP_CHECK) # For some reason there is an error here with the CHECK 
```
```suggestion
vcpkg_fixup_pkgconfig() 
```

Neumann-A commented on ports/dmx/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    REF  6056db9a2fa8ad1ea55f8b8e2cbf5972408d612f #1.1.4
+    SHA512 f7b0a3fb26bc68e5dd27a0afa98ed29fed31956fd07f89b57171d7f9d9a0a87185876551dbf312b8d90a66fa50de06992cf9eb386fa98dd8133946de3c37e274
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/dmx/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
+) 
+#file(TOUCH ${SOURCE_PATH}/m4/dummy)
```
```suggestion
```

Neumann-A commented on ports/dmx/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS 
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/fontenc/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+    PATCHES configure.ac.patch
+            build.patch #patch name
+) 
+#file(TOUCH ${SOURCE_PATH}/m4/dummy)
```
```suggestion

```

Neumann-A commented on ports/fontenc/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS 
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/fs/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/getopt-win32/CONTROL:
```diff
@@ -1,5 +1,5 @@
 Source: getopt-win32
-Version: 0.1
+Version: 0.1-1
```
```suggestion
Version: 0.1
Port-Version: 1
```

Neumann-A commented on ports/ice/portfile.cmake:
```diff
@@ -0,0 +1,35 @@
+    REF 8e6a14c63d6b73cde87cb331439f2a4d19cba5b9 # 1.0.10
+    SHA512  ad79cfbc3b1d51fb1f019bc088999ac8a64062a71667dbb4ffb62fe6d1b7dba7665944f64be6dcd27de08cc77e91512de97231db1e4ac018088727e90113d040
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/ice/portfile.cmake:
```diff
@@ -0,0 +1,35 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/libinput/portfile.cmake:
```diff
@@ -0,0 +1,26 @@
+    REF 5b500c0e744db962fceadccd029fd5075654f4bc # 1.3.4 
+    SHA512  fb89248ca67996fe1b5bd42162153b7b913ced2a7f85b8f3b1e9feb2f1a298907d4a04c3b3cc67417553c8f4e82f74454e3febf5ec187e16969456bb89c55d60
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/libinput/portfile.cmake:
```diff
@@ -0,0 +1,26 @@
+    #PATCHES example.patch #patch name
+) 
+
+set(ENV{ACLOCAL} "aclocal -I \"${CURRENT_INSTALLED_DIR}/share/xorg/aclocal/\"")
```
```suggestion

```

Neumann-A commented on ports/libinput/portfile.cmake:
```diff
@@ -0,0 +1,26 @@
+            -Ddocumentation=false
+)
+vcpkg_install_meson()
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES pthread)
```
```suggestion
vcpkg_fixup_pkgconfig()

```

Neumann-A commented on ports/mtdev/portfile.cmake:
```diff
@@ -0,0 +1,31 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    OPTIONS xorg_cv_malloc0_returns_null=yes
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
    OPTIONS xorg_cv_malloc0_returns_null=yes
```

Neumann-A commented on ports/pciaccess/portfile.cmake:
```diff
@@ -0,0 +1,38 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    OPTIONS
```
```suggestion

```

Neumann-A commented on ports/pthread-stubs/portfile.cmake:
```diff
@@ -0,0 +1,62 @@
+    #PATCHES example.patch #patch name
+) 
+
+#set(ENV{ACLOCAL} "aclocal -I \"${CURRENT_INSTALLED_DIR}/share/xorg/aclocal/\"")
```
```suggestion

```

Neumann-A commented on ports/pthread-stubs/portfile.cmake:
```diff
@@ -0,0 +1,62 @@
+    REF 50f0755a7f894acae168f19c66e52a3f139ca4ec # 0.4.0
+    SHA512  15fcb2144a8abb7b9b1b8f6d9732759351268fb440c7a59380b0ca6ddf48b74a37ce5afbf777ce58fc1993df0c8d6ffb82e452800ce2fcaf16edcbcc1750e338
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/pthread-stubs/portfile.cmake:
```diff
@@ -0,0 +1,62 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/sm/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+    REF a52c79544fcd6b5e2242b9122dfaa34be07aebb2 # 1.2.3
+    SHA512  379e450d90e61d80d4fea8449a582b3eee3968bef137022053cb3bd51fa2815d8fccc43ff11e3b593c4a67ad64e93209c25111a20ac88e38c1f663cd274f5d56
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/sm/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/wayland-protocols/portfile.cmake:
```diff
@@ -0,0 +1,42 @@
+    REF  630fb089103a1d0eab1a684b853ab5c4d2b252aa #1.18
+    SHA512 b9132ebe4a85556cdbf03bab2f874949b004f361c654c115b8a70ce25e582b153fbad7ff4b1b2206d456d5f8222f08129c002c7b80ae56495a5417d7b8bda0b1
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/wayland-protocols/portfile.cmake:
```diff
@@ -0,0 +1,42 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/xau/portfile.cmake:
```diff
@@ -0,0 +1,40 @@
+    REF d9443b2c57b512cfb250b35707378654d86c7dea # 1.0.9 
+    SHA512  d46d2be838b8ea7116ee0e312adafa80f6551762b5f7b5e503bd40e74fc0b43b45255e8135d03f831d5c483b98aac992fcd91a7e22119261e76778571a72ef07
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xau/portfile.cmake:
```diff
@@ -0,0 +1,40 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    OPTIONS ${OPTIONS}
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
    OPTIONS ${OPTIONS}
```

Neumann-A commented on ports/xaw/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xaw/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES pthread)
```
```suggestion
vcpkg_fixup_pkgconfig()
```

Neumann-A commented on ports/xaw3d/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+    REF 62e18b8884efd39e8d764442f7b354d3e34fefbd # 1.6.3 
+    SHA512  c4cfbf3e6085c1b9e0fae3a15b1ec2a27af20c5f14c20a47f4c0fc96512feded888c7152484667135880bf93a12b0837730f29be416ea290795b385cbe511c5e
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/xaw3d/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xaw3d/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES pthread)
```
```suggestion
vcpkg_fixup_pkgconfig()
```

Neumann-A commented on ports/xbitmaps/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+    REF  61eebdfde170841ae933cf65ba27346fbf2f8018 #1.1.2
+    SHA512 dd8acefc6f96d65e4b5d4807221aa3b87faca7b8b50e9de76081351503c9abb127a3063f3bb23ca71a0e3521640e7fdf7686e33cca5c7b7ad8d67a7ac26e65b5
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xbitmaps/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xcb-cursor/portfile.cmake:
```diff
@@ -0,0 +1,30 @@
+    REF  95b9a8fd876fdbbc854cdf3d90317be3846c7417 #0.1.3
+    SHA512 cca7bf1f2aeaab8d256052a676098d7c600b90dc47cf9bc84d11229e59fbf5c83f7f877b8538f7cc662983807566d28c87b3501abc7cab76cc553d9db29eceb9
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xcb-image/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xcb-keysyms/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xcb-render-util/portfile.cmake:
```diff
@@ -0,0 +1,39 @@
+    REF  0317caf63de532fd7a0493ed6afa871a67253747 #v 0.3.9
+    SHA512 9a16c3eb3575586fce85a548cb5fd8b23aa8e42363b2c6d4e074e45f5c04089301ef2ff3012cf9cc94490359d4e28ff062feaecb97a880c9240bf7dbe5d03612
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xcb-render-util/portfile.cmake:
```diff
@@ -0,0 +1,39 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xcb-render-util/portfile.cmake:
```diff
@@ -0,0 +1,39 @@
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES Ws2_32 ws2_32)
```
```suggestion
vcpkg_fixup_pkgconfig()
```

Neumann-A commented on ports/xcb-util-m4/portfile.cmake:
```diff
@@ -0,0 +1,16 @@
+    REF  f662e3a93ebdec3d1c9374382dcc070093a42fed #v1.19.2
+    SHA512 29840da449a434f169437fd2cef78273e0cba00a7f76d48790c838dc8f40fe55cb0932d96b649e1bd066c6c5e257dd2d9d71c663ce100aa5ca25a2ccec1b7e77
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xcb-util/CONTROL:
```diff
@@ -0,0 +1,10 @@
+Homepage: https://xcb.freedesktop.org/
+Description: C interface to the X Window System protocol, which replaces the traditional Xlib interface.
+Build-Depends: xorg-macros, xcb-util-m4, xcb
+Default-Features:
+
+Feature: no-feature
+Description: just a dummy
+Build-Depends:
```
```suggestion
```

Neumann-A commented on ports/xcb-util/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xcb-wm/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xcb/portfile.cmake:
```diff
@@ -0,0 +1,85 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    OPTIONS ${OPTIONS}
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
    OPTIONS ${OPTIONS}
```

Neumann-A commented on ports/xcb/portfile.cmake:
```diff
@@ -0,0 +1,85 @@
+)
+
+vcpkg_install_make(MAKE_OPTIONS -k --print-data-base)
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES Ws2_32 ws2_32)
```
```suggestion
vcpkg_fixup_pkgconfig()
```

Neumann-A commented on ports/xcomposite/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+    REF fd7d02cc014ac9bf5bb7e68b66102ea36a76a59a # 0.4.5
+    SHA512  68ee9d61d364816bf190db010f624312e9daf8d6b411dc209486ac0c7d92b3c75ca5a2650bc085b0c444bec1adfefca593e1d9e6fda4ba12ab5f12ebe7fd6ade
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/xcomposite/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xcomposite/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES pthread)
```
```suggestion
vcpkg_fixup_pkgconfig()
```

Neumann-A commented on ports/xcursor/portfile.cmake:
```diff
@@ -0,0 +1,38 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xcwm/portfile.cmake:
```diff
@@ -0,0 +1,41 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xdamage/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xdamage/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES pthread)
```
```suggestion
vcpkg_fixup_pkgconfig()
```

Neumann-A commented on ports/xdamage/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+    REF 977b04cd69738806e0b48fcf5c725763d065f06d # 1.1.5
+    SHA512  79c3a4c63f6c50c39d324183b98ad7e70235aed1c8385acf2f593739c71e7929119448be3e15dffd276b32e4fbb056508deeb35f450f74b85c101047f68d4339
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xdmcp/portfile.cmake:
```diff
@@ -0,0 +1,37 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    OPTIONS ${OPTIONS} --enable-unit-tests=no
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
    OPTIONS ${OPTIONS} --enable-unit-tests=no
```

Neumann-A commented on ports/xext/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+    REF ebb167f34a3514783966775fb12573c4ed209625 # 1.3.4 
+    SHA512  509cb67ad9dba37adbcc49dca6e9803b318f62e8e2b981ccf73ce328a8104e94615312c5914b480ba04e28d208cd93d8a71b3e6e2ef1a4e5615aa2a3f9ba19e7
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/xext/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL "export ACLOCAL=\"aclocal -I ${CURRENT_INSTALLED_DIR}/share/xorg-macros/aclocal/\""
+    OPTIONS xorg_cv_malloc0_returns_null=yes
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
    OPTIONS xorg_cv_malloc0_returns_null=yes
```

Neumann-A commented on ports/xext/portfile.cmake:
```diff
@@ -0,0 +1,34 @@
+)
+
+vcpkg_install_make()
+vcpkg_fixup_pkgconfig(SYSTEM_LIBRARIES pthread)
```
```suggestion
vcpkg_fixup_pkgconfig()
```

Neumann-A commented on ports/xf86-drivers/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    REF  cfde5e9f348db6e6b2639dd7e592520dff028452
+    SHA512 cca7bf1f2aeaab8d256052a676098d7c600b90dc47cf9bc84d11229e59fbf5c83f7f877b8538f7cc662983807566d28c87b3501abc7cab76cc553d9db29eceb9
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xf86-drivers/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/xf86-input-evdev/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    REF  67a97afbc011799e0089f88f562586a33a312684
+    SHA512 9594aa1a6cc6ef01146ae9f7887f046f7084edddd92f0fc69d7fa45af9043acf480d7886fcd5bf66a7523d98654ecfc8fcf0bb78500907cfc9a37ae2d8be9605
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xf86-input-evdev/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xf86-input-joystick/portfile.cmake:
```diff
@@ -0,0 +1,28 @@
+    REF  de424865ece983d8da62b7f7e3d013eabac0b8b5
+    SHA512 cc2fd0e13588ed531cc369e068cfcbd5aa3d274acc7d0a4543ccf18ecc6a32252060d8ae88de606671d7406b6934704ea0f356b722ea342d239f4c7f7503d7cc
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xf86-input-keyboard/portfile.cmake:
```diff
@@ -0,0 +1,35 @@
+    REF  0c7f51291f91eee61fcaf05080001a5e9b22fb53 #1.9.0
+    SHA512 bc0dcdd932ccc473ca7182523561ed5a777b5be385bdec1d091bf0e48124a1bb22a29ec47d15f72a837d7dfb41adb94c7d1cee6a1c0157c4dc02224f8a47d64c
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/xf86-input-keyboard/portfile.cmake:
```diff
@@ -0,0 +1,35 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xf86-input-libinput/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    REF  583ed5acdc17a4dc7848c891d7cdbd35ff07b781 #0.30.0
+    SHA512 7ee8faff3f7670b5597d63e0b6b249ef555b9899ee0fd447b4533104e20fc15b741a23a851cda83bbfefcc02216748bf3ef6b5f3f1be0cdcc5a3d3de74e8fcd8
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/xf86-input-libinput/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xf86-input-mouse/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    REF  cfde5e9f348db6e6b2639dd7e592520dff028452
+    SHA512 4e0803c7ce82c3369d2cb02018c59cb72c242f5b35cea7c8f330666e02f0e9de48e5b5741b0ab7cb7ab4bacb8483021526f5b4f2e520eaa9134140bab54a9e35
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/xf86-input-mouse/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xf86-video-amdgpu/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    REF  b467d2569a003da05ad222b0dc095bee5eec450a
+    SHA512 76166f0658f8075eac3e9dd133d4508fc2c4eea73a59fb7043a67f67d9db80b77b10e239ce1bf3c7555ad3e4a3f3750b36931350c4da8d1b2a1ea730f1d6c5e7
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xf86-video-amdgpu/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/xf86-video-ati/portfile.cmake:
```diff
@@ -0,0 +1,35 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A commented on ports/xf86-video-fbdev/portfile.cmake:
```diff
@@ -0,0 +1,35 @@
+    REF  82aa13d01340236ebb516ea45a9bd78e4fe52b79
+    SHA512 16cc97a1f761b38d4aba389ff4e59dfa8e985d0d3b545a4348f42e1aff44e926440638d80fb7f58b2aa01e3dfadce1727846824801b295bdd4444c8cafaa3d9e
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/xf86-video-fbdev/portfile.cmake:
```diff
@@ -0,0 +1,35 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/xf86-video-intel/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    REF  baec802b21387d04aebb10ac29e719a1800c5aa0
+    SHA512 47e5cc7ca93cb948d0222dec7627ab1a07e1ad451114a2cc732dcb3f06580237db2c5e70fb12d1c39255d426ed22044e5a4fe8433725f25471665a1b2d9e6146
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/xf86-video-intel/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion

```

Neumann-A commented on ports/xf86-video-nouveau/portfile.cmake:
```diff
@@ -0,0 +1,35 @@
+    REF  ec2b45d1bf4c31c5420b48c213bf9ee1656adf9d
+    SHA512 547be83e31ca3a9bb538ec0f0fa320e804c7164d6261fd2889e13c6ffe6e21887fda90bc9534eab551ec544172b30a4363c041eb7788001b6ba5692c43d824e7
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion
```

Neumann-A commented on ports/xf86-video-nv/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+    REF  e4134c4ebd3522e730689e0bda7b9f8fd39aedb8
+    SHA512 848d340472c966a645a3d8eb9637c1b333d02ad9a273c95b85bc844d04c02378ce616ffb200ea2cadb9b0f40bbf9e0a529626697015be5d8e08333588b5eb789
+    HEAD_REF master # branch name
+    #PATCHES example.patch #patch name
```
```suggestion

```

Neumann-A commented on ports/xf86-video-nv/portfile.cmake:
```diff
@@ -0,0 +1,36 @@
+vcpkg_configure_make(
+    SOURCE_PATH ${SOURCE_PATH}
+    AUTOCONFIG
+    #SKIP_CONFIGURE
+    #NO_DEBUG
+    #AUTO_HOST
+    #AUTO_DST
+    #PRERUN_SHELL ${SHELL_PATH}
+    #OPTIONS
+    #OPTIONS_DEBUG
+    #OPTIONS_RELEASE
```
```suggestion
```

Neumann-A committed:
Cleanup C&P Code

Neumann-A committed:
more cleanup

Neumann-A committed:
add lint feature back in

Neumann-A committed:
build xmllint in libxml2

Neumann-A committed:
dont't change line endings on patch

Neumann-A committed:
update xcb-proto

Neumann-A committed:
fix cpp wrapper to work.

Neumann-A committed:
remove libtool artifacts

Neumann-A committed:
fix wrapper
use old checkout
try debugging the build by adjusting the wrappers.

Neumann-A committed:
fix xlib dynmic windows build
update to 1.7.0 again

Neumann-A committed:
fix xcb-render-util

Neumann-A committed:
fix windows xfont build

Neumann-A committed:
need to remove extra whitespaces in c(pp) args

Neumann-A committed:
update xserver

Neumann-A committed:
add some supports fields.

Neumann-A committed:
more supports.

Neumann-A committed:
copy tool deps

Neumann-A committed:
add todo for xwindow

Neumann-A committed:
fix xcursorgen deps

Neumann-A committed:
Merge remote-tracking branch 'RichLogan/meson-rev' into add_x_window_system

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

# Conflicts:
#	scripts/cmake/vcpkg_build_make.cmake

Neumann-A committed:
fix details

Neumann-A committed:
fix xres on windows

Neumann-A committed:
Merge branch 'add_x_window_system' of https://github.com/Neumann-A/vcpkg into add_x_window_system

ras0219 commented on ports/munit/CONTROL:
```diff
@@ -5,3 +5,4 @@ Description: A small but full-featured unit testing framework for C
 Build-Depends: tool-meson
 Supports: !(arm|arm64|uwp)
 Homepage: https://github.com/nemequ/munit
+Build-Depends: tool-meson
```
Duplicate of line 5

Neumann-A committed:
fix windows regressions

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

Neumann-A committed:
remove double build-depends

Neumann-A committed:
reset some changes to upstream

Neumann-A committed:
Merge remote-tracking branch 'upstream/master' into add_x_window_system

# Conflicts:
#	ports/libxml2/CONTROL
#	ports/libxml2/portfile.cmake
#	ports/pixman/CONTROL

JackBoosY commented:
This PR has been inactive for a long time, please reopen it if there is any progress.

Neumann-A commented:
@JackBoosY This PR just requires a word on how to deal with it on linux systems where x libs are normally provided by the system

JackBoosY commented:
Should I reopen this? Should I give you some advice? But first you need to provide the current situation.

Neumann-A commented:
@JackBoosY 

> Should I reopen this? Should I give you some advice? But first you need to provide the current situation.

So the problem is as follows: 
- On Linux X is typically provided by the system. Should the port in vcpkg be empty in this case or should we switch it to dynamic linkage in this case? Or should there be a way to deactivate X for the user by setting some variable?
- A lot of ports depend on X but don't declare that dependency. I will probably not change any other ports to declare that dependency correctly considering the waste amount of ports. This needs to be done by other maintainers on port update. 
- There is probably more to discuss but these are the biggest points which need to be solved first.

JackBoosY commented:
> On Linux X is typically provided by the system. Should the port in vcpkg be empty in this case or should we switch it to dynamic linkage in this case? Or should there be a way to deactivate X for the user by setting some variable?

All libraries provided only by the system should be set to empty ports, and search and use functions should be provided in them. Such as CUDA.

> A lot of ports depend on X but don't declare that dependency. I will probably not change any other ports to declare that dependency correctly considering the waste amount of ports. This needs to be done by other maintainers on port update.

Yes, for those ports that use X libraries, they should explicitly state that they depend on X libraries, and these modifications can be done during the update. You can also temporarily disable these features that depends on X libraries.
